src/system_data.jl

```diff
@@ -811,3 +818,54 @@ end
 
 _get_system_basename(system_file) = splitext(basename(system_file))[1]
 _get_secondary_basename(system_basename, name) = system_basename * "_" * name
+
+add_supplemental_attribute!(data::SystemData, component, info; kwargs...) =
+    add_supplemental_attribute!(data.attributes, component, info; kwargs...)
+
+function get_supplemental_attributes(
+    filter_func::Function,
+    ::Type{T},
+    data::SystemData,
+) where {T <: InfrastructureSystemsSupplementalAttribute}
+    return get_supplemental_attributes(T, data.attributes, filter_func)
+end
+
+function get_supplemental_attributes(
+    ::Type{T},
+    data::SystemData,
+) where {T <: InfrastructureSystemsSupplementalAttribute}
+    return get_supplemental_attributes(T, data.attributes)
+end
+
+function iterate_supplemental_attributes(data::SystemData)
+    return iterate_supplemental_attributes(data.attributes)
+end
+
+function remove_supplemental_attribute!(
+    data::SystemData,
+    info::T,
+) where {T <: InfrastructureSystemsSupplementalAttribute}
+    current_components_uuid = collect(get_component_uuids(info))
+    for c_uuid in current_components_uuid
+        component = get_component(data, c_uuid)
+        delete!(get_supplemental_attributes_container(component), info)
+        delete!(get_component_uuids(info), get_uuid(component))
+    end
+
+    return remove_supplemental_attribute!(data.attributes, info)
+end
+
+function remove_supplemental_attributes!(
```

Comment:
I have concerns about this function because of the implicit disconnection from components. Since the user can only add attributes through a component, shouldn’t they only be able to remove attributes through a component?

Reply:
This contradicts previous requests to have a method to do this

Reply:
This method will clear all attributed of a particular type